core/model/impl/src/main/kotlin/FeatureModelImpl.kt

```diff
@@ -71,43 +77,53 @@ internal class FeatureModelImpl private constructor(
         }
     }
 
+    private val conditionExpressionHolder: ConditionExpressionHolder? by lazy {
+        impl.getAnnotation(BuiltinAnnotation.ConditionExpression)?.let { ConditionExpressionHolder(it) }
+    }
+
     private fun parseOneCondition(one: BuiltinAnnotation.ConditionFamily.One): BooleanExpressionInternal {
-        val match = ConditionRegex.matchEntire(one.condition)
-        return if (match != null) {
-            val (negate, names) = match.destructured
-            val variable = ConditionModelImpl(
-                type = one.target,
-                pathSource = names,
-            )
-            if (negate.isNotEmpty()) {
-                NotExpressionImpl(variable)
-            } else variable
+        val condition = one.condition
+        return if (condition[0] == '!') {
```

Comment:
is condition.length > 0 is checked somwhere before?

Reply:
Yeah, it's not checked, gotta fix that.